components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1505,6 +1507,76 @@ public static SpOAuth2ExpiryTimeConfiguration getSpTokenExpiryTimeConfig(String
         return spTokenTimeObject;
     }
 
+    /**
+     * Retrieve audience configured for the particular service provider.
+     *
+     * @param clientId
+     * @param oAuthAppDO
+     * @return
+     */
+    public static List<String> getOIDCAudience(String clientId, OAuthAppDO oAuthAppDO) {
+
+        List<String> oidcAudiences = getDefinedCustomOIDCAudiences(oAuthAppDO);
+        // Need to add client_id as an audience value according to the spec.
+        if (!oidcAudiences.contains(clientId)) {
+            oidcAudiences.add(0, clientId);
+        } else {
+            Collections.swap(oidcAudiences, oidcAudiences.indexOf(clientId), 0);
+        }
+        return oidcAudiences;
+    }
+
+    private static List<String> getDefinedCustomOIDCAudiences(OAuthAppDO oAuthAppDO) {
+
+        List<String> audiences = new ArrayList<>();
+
+        // Priority should be given to service provider specific audiences over globally configured ones.
+        if (OAuth2ServiceComponentHolder.isAudienceEnabled()) {
+            List<String> audienceList = Arrays.asList(oAuthAppDO.getAudiences());
+            audiences = new ArrayList<>(audienceList);
```

Comment:
Why do we have this again since the audienceList is String list?

Reply:
Answer: 
since we want to add one element to the audiences list in the method `getOIDCAudience(String clientId, OAuthAppDO oAuthAppDO)`
 else it will throw "java.lang.UnsupportedOperationException" refer:
https://stackoverflow.com/questions/5755477/java-list-add-unsupportedoperationexception